### DISCUSSION TITLE
How do you connect to the local running Postgres instance?

### DISCUSSION THREAD

**Discussion by jvz** [pgjdbc/r2dbc-postgresql#577] [Q&A]
Using the JDBC driver, this would normally work via a trivial JDBC URL like `jdbc:postgresql:database`. I can't figure out the equivalent connection string for the R2DBC driver. I get an error about a username not being specified (there isn't a username or password to authenticate with; this is the Postgres running as the same user).

**Comment by jvz** [comment]
I'll note that I can sort of work around this by setting the connection URL to `r2dbc:postgresql://localhost/database` and setting the `user` option to my username. This seems to use the trust authentication mechanism over TCP. The equivalent socket file for this is `/tmp/.s.PGSQL.5432` by default, though I'm not sure how you can create a connection URL using that.

**Reply by mp911de** [reply]
The URL counterpart would look like: `r2dbc:postgresql:///foo?socket=/tmp/.s.PGSQL.5432` (or `r2dbc:postgresql://postgres@/foo?socket=/tmp/.s.PGSQL.5432` when specifying an user).

**Reply by jvz** [reply]
Sweet, thanks!

**Reply by jvz** [reply]
I'm assuming the JDBC driver must auto-fill the `user.name` system property as the user when connecting if none is specified. Well, combined with the R2DBC URI here, I've got things working for development use, too, so thanks!
